File: API/DTOs/OrderDto.cs
﻿using System.ComponentModel.DataAnnotations;
using API.Entity.OrderAggregate;

namespace API;

public class OrderDto
{
    public int Id { get; set; }
    public required string BuyerId { get; set; }

    public required ShippingAddress ShippingAddress { get; set; }
    public DateTime OrderDate { get; set; }
    public required List<OrderItemDto> OrderItems { get; set; }
    public long Subtotal { get; set; }
    public long DeliveryFee { get; set; }
    public required string OrderStatus { get; set; }
    public long total { get; set; }


}

File: client/src/app/models/orderDto.ts
import { BasketItem } from "./basket";

export interface orderDto {
  id: number;
  buyerId: number;
  shippingAddress: ShippingAddress;
  orderDate: string;
  orderItems: BasketItem[];
  subtotal: number;
  deliveryFee: number;
  orderStatus: string;
  total: number;
}

export interface ShippingAddress {
  fullName: string;
  address1: string;
  address2: string;
  city: string;
  state: string;
  zip: string;
  country: string;
}
